packages/react-sdk/src/components/components-utils.ts

```diff
@@ -0,0 +1,86 @@
+import * as components from "./components";
+import { registeredComponents } from "./index";
+
+export type ComponentName = keyof typeof components;
+
+/**
+ * We need to define component names manually here, instead of using Object.keys(components)
+ * Otherwise every component would be included in the bundle, even if it is not used
+ */
+const componentNames = [
+  "Box",
+  "Body",
+  "TextBlock",
+  "Heading",
+  "Paragraph",
+  "Link",
+  "RichTextLink",
+  "Span",
+  "Bold",
+  "Italic",
+  "Superscript",
+  "Subscript",
+  "Button",
+  "Input",
+  "Form",
+  "Image",
+  "Blockquote",
+  "List",
+  "ListItem",
+] as const;
+
+type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y
```

Comment:
oh nooooo 😭 is there a way to not have it? @Andarist

Reply:
This type checks type "identity", so for example those won't be treated as the same types:
```ts
type A = { a: number, b: string }
type B = { a: number } & { b: string }
```

I'd say that usually it's not what you actually want, so this can be rewritten like this:
```ts
type Equals<A, B> = [A] extends [B] ? [B] extends [A] ? true : false : false;
```
Or even like this:
```ts
type Equals<A, B> = [A, B] extends [B, A] ? true : false;
```
Although I've seen the latter not working in some quite complex generic cases (so the problem is likely not applicable to you + it's a TS bug)